### PR TITLE
Vectorize return values of operators

### LINKED ISSUE #107: Behavior of 0-dimensional array
Behavior of 0-dimensional array is strange.

```
>>> x = numpy.array(1)
>>> x
array(1)
>>> x + x
2
```

So, infix operators of `Variable` also have the same problem.

```
>>> x = chainer.Variable(numpy.array(1))
>>> x.data
array(1)
>>> y = x + x
>>> y.data
2
```

We have two choices:

- Forbid 0-dimensional array. Instead use 1-dimension array which has only one element.
- Check type of an array in `Function`s. When an array is converted to a scalar, we need to re-convert the value to 0-dimensional array.

This issue is related to #100

### PR DESCRIPTION
**Pull request by unnonouno**
I wrapped scalar values with 0-dimensional vector.

fix #107

### CODE REVIEW

**Comment by beam2d** [comment]
I'll check.

**Comment by beam2d** [comment]
Add _vectorize to Neg, too.

**Comment by unnonouno** [comment]
woops

**Comment by unnonouno** [comment]
I found that `Neg` is not tested. https://coveralls.io/builds/2937341/source?filename=chainer%2Ffunctions%2Fbasic_math.py#L29

File: chainer/functions/basic_math.py [hunk 1]
```diff
@@ -12,6 +12,16 @@
 # Arithmetic
 # ------------------------------------------------------------------------------
 
+def _vectorize(x, typ):
```
Comment (beam2d) [inline chainer/functions/basic_math.py]: Should we move it to some public place as a utility? Other functions may want to use it.

File: chainer/functions/basic_math.py [hunk 2]
```diff
@@ -12,14 +12,24 @@
 # Arithmetic
 # ------------------------------------------------------------------------------
 
+def _vectorize(x):
```
Comment (beam2d) [inline chainer/functions/basic_math.py]: We might want to use this function from other functions. Should we move it to some public place, e.g. chainer.utils?

**Comment by beam2d** [comment]
LGTM except one comment!

File: chainer/functions/basic_math.py (hunk 2 above)
Comment (unnonouno) [inline chainer/functions/basic_math.py]: OK. How about the name? `chainer.utils.vectorize`?

Comment (beam2d) [inline chainer/functions/basic_math.py]: "vectorize" may be a bit confusing name. How about `chainer.utils.force_array`?

Comment (unnonouno) [inline chainer/functions/basic_math.py]: It sounds good!

**Comment by beam2d** [comment]
[LGTM](http://lgtm.in/i/kycNJ9leq)

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**